parla/ldevice.py

```diff
@@ -9,10 +9,12 @@
 from abc import abstractmethod, ABCMeta
 from functools import reduce
 from math import floor, ceil
-from typing import List, Tuple, Collection, Mapping
+from typing import List, Tuple, Iterable, Collection, Mapping
 from warnings import warn
 
 from parla.device import get_all_devices, Device, Memory, MemoryKind
+from parla.tasks import get_placement_for_any # TODO (bozhi): move get_placement_for_xxx to parla.device?
```

Comment:
I suspect `placement` will need it's own module. It is an interface and cuts across multiple parts of the the system. You can use `__parla_placement__` to avoid requiring the `placement` module to import a bunch of stuff.

Reply:
Are `placement` and `device` the same concept in Parla? We already have a `device` module and the `get_placement_for_any/set/value` interfaces ultimately use `__parla_placement__` for some types.

Reply:
A placement is a collection of devices, interpreted as a disjunction (A or B or C). Not just a device. Or could go in device.py though. It's a pretty simple abstraction. Up to you.

Reply:
Let me put these placement-related interfaces into the `device` module for now and maybe further rename the module to be the broader `placement`. After all, I think it anyways makes more sense than nesting them in `tasks`.

Reply:
Leave it an issue #25.